modules/mesh-task/iam.tf

```diff
@@ -12,8 +12,8 @@ locals {
   // it into a single-element list...If the value is null then the splat expression will return
   // an empty tuple."
   // https://www.terraform.io/docs/language/expressions/splat.html#single-values-as-lists
-  create_task_role      = length(var.task_role[*]) == 0
-  create_execution_role = length(var.execution_role[*]) == 0
+  create_task_role      = length(var.task_role[*]) == 0 ? true : length(var.task_role.id[*]) == 0
+  create_execution_role = length(var.execution_role[*]) == 0 ? true : length(var.execution_role.id[*]) == 0
```

Comment:
Terraform conditionals do not short circuit, so using `length(var.task_role[*]) == 0 || length(var.task_role.id[*]) == 0` will error when `var.task_role = null`. Hence, the workaround.